app/src/main/java/com/duckduckgo/app/di/StatisticsModule.kt

```diff
@@ -16,25 +16,51 @@
 
 package com.duckduckgo.app.di
 
+import com.duckduckgo.app.global.AppUrl
+import com.duckduckgo.app.global.api.ApiRequestInterceptor
 import com.duckduckgo.app.statistics.VariantManager
+import com.duckduckgo.app.statistics.api.PixelService
 import com.duckduckgo.app.statistics.api.StatisticsRequester
 import com.duckduckgo.app.statistics.api.StatisticsService
 import com.duckduckgo.app.statistics.api.StatisticsUpdater
+import com.duckduckgo.app.statistics.pixels.ApiBasedPixel
+import com.duckduckgo.app.statistics.pixels.Pixel
 import com.duckduckgo.app.statistics.store.StatisticsDataStore
 import dagger.Module
 import dagger.Provides
+import okhttp3.OkHttpClient
 import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
 
 
 @Module
 class StatisticsModule {
 
     @Provides
-    fun statisticsService(retrofit: Retrofit): StatisticsService =
-        retrofit.create(StatisticsService::class.java)
+    fun statisticsService(retrofit: Retrofit): StatisticsService = retrofit.create(StatisticsService::class.java)
 
     @Provides
     fun statisticsUpdater(statisticsDataStore: StatisticsDataStore, statisticsService: StatisticsService, variantManager: VariantManager) : StatisticsUpdater =
         StatisticsRequester(statisticsDataStore, statisticsService, variantManager)
 
+    @Provides
+    fun pixelService(apiRequestInterceptor: ApiRequestInterceptor) : PixelService {
+
+        // ideally these would be injected but @Named parameters seem to break dagger
```

Comment:
Curious to know more about this - `@Named` should work ok

Reply:
Thanks @CDRussell ! I took a second look and got it to work :)

The error message I was getting wasn't clear, it was complaining about something not being marked as `@Nullable` along with a bunch of other Dagger guff, but the actual error was that I hadn't annotated every place the named instances were being injected.